src/chunk_constraint.c

```diff
@@ -451,34 +482,52 @@ chunk_constraint_dimension_slice_id_tuple_found(TupleInfo *ti, void *data)
  * constraints are saved in the chunk scan context.
  */
 int
-chunk_constraint_scan_by_dimension_slice_id(DimensionSlice *slice, ChunkScanCtx *ctx)
+chunk_constraint_scan_by_dimension_slice(DimensionSlice *slice, ChunkScanCtx *ctx)
 {
-	Catalog    *catalog = catalog_get();
 	ScanKeyData scankey[1];
 	ChunkConstraintScanData data = {
 		.scanctx = ctx,
 		.slice = slice,
 	};
-	ScannerCtx	scanctx = {
-		.table = catalog->tables[CHUNK_CONSTRAINT].id,
-		.index = catalog->tables[CHUNK_CONSTRAINT].index_ids[CHUNK_CONSTRAINT_CHUNK_ID_DIMENSION_SLICE_ID_IDX],
-		.scantype = ScannerTypeIndex,
-		.nkeys = 1,
-		.scankey = scankey,
-		.data = &data,
-		.filter = chunk_constraint_for_dimension_slice,
-		.tuple_found = chunk_constraint_dimension_slice_id_tuple_found,
-		.lockmode = AccessShareLock,
-		.scandirection = ForwardScanDirection,
-	};
 
 	ScanKeyInit(&scankey[0],
 				Anum_chunk_constraint_chunk_id_dimension_slice_id_idx_dimension_slice_id,
 				BTEqualStrategyNumber,
 				F_INT4EQ,
 				Int32GetDatum(slice->fd.id));
 
-	return scanner_scan(&scanctx);
+	return chunk_constraint_scan_internal(CHUNK_CONSTRAINT_CHUNK_ID_DIMENSION_SLICE_ID_IDX,
+										  scankey,
+										  1,
+										  chunk_constraint_dimension_slice_id_tuple_found,
+										  chunk_constraint_for_dimension_slice,
+										  &data,
+										  AccessShareLock);
+}
+
+/*
+ * Scan for chunk constraints given a dimension slice ID.
+ *
+ * Optionally, collect all chunk constraints if ChunkConstraints is non-NULL.
+ */
+int
+chunk_constraint_scan_by_dimension_slice_id(int32 dimension_slice_id, ChunkConstraints *ccs)
```

Comment:
can't we unify with `chunk_constraint_scan_by_dimension_slice` to reduce code dup?

Reply:
done